Source/WebCore/page/InteractionRegion.cpp

```diff
@@ -441,37 +463,37 @@ std::optional<InteractionRegion> interactionRegionForRenderedRegion(RenderObject
     RefPtr styleClipPath = regionRenderer.style().clipPath();
 
     if (styleClipPath && styleClipPath->type() == PathOperation::OperationType::Shape && originalElement) {
-        auto boundingRect = absoluteBoundingRect(regionRenderer);
-        clipPath = styleClipPath->getPath(TransformOperationData(FloatRect(FloatPoint(), boundingRect.size())));
+        auto size = boundingSize(regionRenderer, transform);
```

Comment:
I don't know if this transform thing is correct. You've deviated so far from the painting code that it's impossible to know how much of this transform/clipping logic is OK.